addons/sale/models/product_product.py

```diff
@@ -34,6 +34,16 @@ def _compute_sales_count(self):
             product.sales_count = float_round(r.get(product.id, 0), precision_rounding=product.uom_id.rounding)
         return r
 
+    @api.onchange('type')
+    def _onchange_type(self):
+        res = super(ProductProduct, self)._onchange_type()
+        if self.sales_count > 0:
```

Comment:
```suggestion
        if self.id and self.sales_count > 0:
```

No need to query orders count for non saved product(s) ;).